### DISCUSSION TITLE
子类初始化父类对象

### DISCUSSION THREAD

**Discussion by 0KEAHA** [liusy58/NJU_AdvancedProgramming_SP22#188] [Q&A]
![QQ截图20220428204854](https://user-images.githubusercontent.com/101489263/165757522-2b609baf-0150-479d-b830-2f9f73c5cecf.jpg)
这里的QList的存储的是BaseCloseDefender指针，BaseCloseDefender继承自BaseDefender，BaseDefender继承自BaseActionUnit
BaseActionUnit继承自QGraphicObject，自身是一个抽象类
GetHealth是BaseActionUnit的方法，在BaseActionUnit中宏定义了一个属性Q_PROPERTY  unsigned Health ,GetHealth是它的READ函数，非虚函数
在子类型使用GetHealth方法时报错

**Comment by 0KEAHA** [comment]
把头文件包含放到了cpp里面，报错消失了，可能是头文件包含设计不够精巧，导致编译器并不能理解类之间的关系

**Reply by liusy58** [reply]
more concrete is better.
